src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/move_readvalue_inputs_to_subgraph.cpp

```diff
@@ -0,0 +1,187 @@
+                            assign = ov::as_type_ptr<ov::op::v6::Assign>(grandson.get_node()->shared_from_this());
+                            if (assign) {
+                                break;
+                            }
+                        }
+                    }
+                    if (assign) {
+                        if (assign->get_variable_id() == readvalue->get_variable_id()) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return false;
+}
+
+ov::intel_cpu::MoveReadValueInputsToSubgraph::MoveReadValueInputsToSubgraph() {
+    MATCHER_SCOPE(MoveReadValueInputsToSubgraph);
+    using namespace ov::pass::pattern;
+
+    auto readvalue_pattern = pass::pattern::wrap_type<ov::op::v6::ReadValue>();
+
+    ov::matcher_pass_callback callback = [=](Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto readvalue = as_type_ptr<ov::opset6::ReadValue>(pattern_map.at(readvalue_pattern).get_node_shared_ptr());
+        if (!readvalue || readvalue->get_input_size() != 1u) {
+            return false;
+        }
+
+        // TODO: Temporarily skip this pattern. If MemoryInputSDPA supports Subgraph in the future, it may be deleted.
+        if (skip_pattern_sdpa(readvalue)) {
+            return false;
+        }
+
+        bool found_output = false;  // Flag: find Output node
+
+        NodeVector subgraph_nodes;
+        std::unordered_set<std::string> subgraph_node_names;
+        std::unordered_set<std::shared_ptr<ov::Node>> visited;  // Visited flag.
+        NodeVector inputs = {};
+        OutputVector outputs = {};
+
+        // DFS, Check if current node's final successor is only ReadValue.
+        std::function<void(std::shared_ptr<ov::Node>)> dfs = [&](std::shared_ptr<ov::Node> node) {
+            if (visited.find(node) != visited.end()) {
+                return;
+            }
+
+            // Output node.
+            if (node->get_output_target_inputs(0).size() == 0u) {
+                found_output = true;
+                return;
+            }
+
+            for (const auto& child : node->get_output_target_inputs(0)) {
+                auto son = child.get_node()->shared_from_this();
+                if (son == readvalue) {
+                    continue;
+                }
+                dfs(son);
+            }
+            if (!found_output) {
+                visited.insert(node);
+            }
+        };
+
+        std::function<void(std::shared_ptr<ov::Node>)> reverse_dfs = [&](std::shared_ptr<ov::Node> node) {
+            if (ov::op::util::is_parameter(node)) {
+                inputs.emplace_back(node);
+                return;
+            }
+
+            // Check whether current node have same successor[root_node_name].
+            found_output = false;
+            dfs(node);
+            visited.insert(node);
```

Comment:
It looks like there is a significant lack in the transformation testing. Please add more transformation tests.
The main issue is that the nodes on the output path and the nodes on the path to ReadValue are inserted into the same `visited` set, so during the subsequent dfs calls, it's impossible to distinguish which type paths the node belongs to. The whole idea  of coloring the graph is using different colors (sets in this case) for the Output path and the path leading to the ReadValue.

Reply:
Sorry, I don't get your concern case.
As you said, I have only one `visited` set, the nodes of path to ReadValue and the nodes of path to Output, they are all put into 'visited' set. 
Could you please give a simple pattern about your concern? 
Because it is complex to verify it in our OpenVINO codes, I have a simple test case to verify it. Just provide pattern.
https://github.com/xipingyan/graph_alg/tree/main/02_node_init_graph, 
For example: https://github.com/xipingyan/graph_alg/blob/main/02_node_init_graph/test_graph_2.cpp#L7-L22

BTW, I think it is special case of color map, we only find inputs nodes(only reverse search, and it is directed graph).

Reply:
1. There is a framework for transformations testing, please refer to: `src/plugins/intel_cpu/tests/unit/transformations`. There we store transformations tests for CPU plugin specific transformations. And such tests are a mandatory requirement for newly introduced transformations. So please add such tests. (using the mentioned pattern as well).
2. I'm not talking about the classical graph coloring problem (when we want to color graph such that no two adjacent vertices are of the same color). I'm talking about coloring (or labeling if you wish) as a technique to reduce the number of subsequent dfs searches. In this problem we have two types of paths in the graph: first - all the paths in the graph for which the only destination is a specific ReadValue node (therefore all the vertices on these paths are rolled into the ReadValue), second - all the paths in the graph which destination is an Output node, so the vertices belonging to such paths may not be fused into ReadValue.
Some nodes may belong to both type of paths simultaneously as they have more then one children, and if the node belong to an Output path, it must be excluded from the fused init subgraph.
Now. We use dfs to identify to which path belongs a specific node. So we may perform a dfs each time, which would end up in O(v^2) complexity, or we may color (or label, or store into a set) all the vertices beloging to paths leading to an Output node during a single dfs, so that for each new unlabeled node it's just enough to check if any of the children is already labeled (or stored into a specific set) as belonging to a path leading to an Output node. Therefore such a node automatically belongs to an Output path thus may not be fused into a ReadValue. On the other hand, there are nodes belonging to an ReadValue path (a path which lead to a specific read value node) and they must be colored (labeled or stored in set) differently, so that when we perform a dfs, we simply check that all the children of a node belong to the ReadValue path, as a result the node also belongs to that path.
In your implementation (in its current state) `visited` set is populated only if the node belongs to a path leading to ReadValue, which doesn't help to amortize subsequent dfs calls (as we have to go up to Output each time). But during reverse dfs, "visited" nodes are inserted to the `visited` set unconditionally (without checking the result of dfs), but the very the same `visited` set is used in the dfs call as a condition to return leaving the `found_output` flag unchanged e.g. `false`, signaling that the node belongs to a ReadValue path. In the end, nodes "visited" during reverse dfs, may be a part of an Output path, but still labeled "visited", so the subsequent dfs calls consider them as belonging to the ReadValue path.

Reply:
Thanks very much. I add 2 hash table to distinguish them. and add associated test in https://github.com/openvinotoolkit/openvino/pull/26130/files#diff-4aea6a7ca0f1be8b98068c79de1de9877c5474e693c08b0ae3f4cd2ac7ffecbc
The second test is very complex pattern, maybe enough, if you still have concern,  please give me some patterns, I can add them quickly.